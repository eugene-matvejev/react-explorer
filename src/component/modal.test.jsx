import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from './modal';

configure({ adapter: new Adapter() });

describe('<Modal/>', () => {
    const props = {
        onClose: jest.fn(),
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Modal {...props} />);

            expect(c).toMatchSnapshot();
        });
        // 'data-cy': PropTypes.string,
        // className: PropTypes.string,
        // onClose: PropTypes.func.isRequired,

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Modal {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
