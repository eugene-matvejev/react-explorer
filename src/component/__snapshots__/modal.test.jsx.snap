// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> render with default/required props 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal "
  >
    <div
      className="modal_title"
    >
      <button
        className="modal-control__close"
        data-cy="-modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal_body"
    />
  </section>
</div>
`;

exports[`<Modal/> render with optional props [::className] as "{{className}}" 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal {{className}}"
  >
    <div
      className="modal_title"
    >
      <button
        className="modal-control__close"
        data-cy="-modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal_body"
    />
  </section>
</div>
`;

exports[`<Modal/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<div
  className="modal_container"
>
  <section
    className="modal "
  >
    <div
      className="modal_title"
    >
      <button
        className="modal-control__close"
        data-cy="{{data-cy}}-modal-close"
        onClick={[MockFunction]}
      >
        ×
      </button>
    </div>
    <div
      className="modal_body"
    />
  </section>
</div>
`;
